lib/membrane_rtc_engine/engine.ex

```diff
@@ -1011,10 +837,12 @@ defmodule Membrane.RTC.Engine do
   #
 
   defp build_track_added_actions(tracks, endpoint_id, state) do
-    Enum.flat_map(state.endpoints, fn
-      {^endpoint_id, _endpoint} -> []
-      {_, nil} -> []
-      {endpoint_id, _} -> [forward: {{:endpoint, endpoint_id}, {:new_tracks, tracks}}]
+    state.endpoints
+    |> Map.delete(endpoint_id)
+    |> Map.keys()
+    |> Enum.reject(fn id -> Map.has_key?(state.peers, id) and is_nil(state.peers[id].metadata) end)
```

Comment:
I think this will work. It's `Enum.reject`, so this reads as "reject an endpoint which is a peer and doesn't have any metadata". Of course, if you register HLS as peer, it will not work, but that's not something that you should do